src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,541 @@
+
+            DPR is a model with absolute position embeddings so it's usually advised to pad the inputs on
+            the right rather than the left.
+
+            Indices can be obtained using :class:`transformers.DPRTokenizer`.
+            See :func:`transformers.PreTrainedTokenizer.encode` and
+            :func:`transformers.PreTrainedTokenizer.convert_tokens_to_ids` for details.
+"""
+
+
+@add_start_docstrings(
+    "The bare DPRContextEncoder transformer outputting pooler outputs as context representations.",
+    DPR_START_DOCSTRING,
+    DPR_ENCODERS_INPUTS_DOCSTRING,
+)
+class DPRContextEncoder(DPRPretrainedContextEncoder):
+    r"""
+    Outputs: The DPR encoder only outputs the `pooler_output` that corresponds to the context representation:
+        **pooler_output**: ``torch.FloatTensor`` of shape ``(batch_size, hidden_size)``
+            Last layer hidden-state of the first token of the sequence (classification token)
+            further processed by a Linear layer. This output is to be used to embed contexts for
+            nearest neighbors queries with question embeddings.
+
+    Examples::
+
+        tokenizer = DPRTokenizer.from_pretrained('dpr-base-uncased')
+        model = DPRContextEncoder.from_pretrained('dpr-ctx_encoder-base')
+        input_ids = torch.tensor(tokenizer("Hello, my dog is cute")).unsqueeze(0)  # Batch size 1
+        embeddings = model(input_ids)  # the embeddings of the given context.
+
+    """
+
+    def __init__(self, config: DPRConfig):
+        super().__init__(config)
+        self.config = config
+        self.ctx_encoder = DPREncoder(config)
+        self.init_weights()
+
+    def forward(
+        self, input_ids: Tensor, token_type_ids: Optional[Tensor] = None, attention_mask: Optional[Tensor] = None
+    ) -> Tensor:
+        if attention_mask is None:
+            attention_mask = input_ids != self.config.pad_token_id
+            attention_mask = attention_mask.to(device=input_ids.device)
+        sequence_output, pooled_output, hidden_states = self.ctx_encoder(input_ids, token_type_ids, attention_mask)
+        return pooled_output
+
+
+@add_start_docstrings(
+    "The bare DPRQuestionEncoder transformer outputting pooler outputs as question representations.",
+    DPR_START_DOCSTRING,
+    DPR_ENCODERS_INPUTS_DOCSTRING,
+)
+class DPRQuestionEncoder(DPRPretrainedQuestionEncoder):
+    r"""
+    Outputs: The DPR encoder only outputs the `pooler_output` that corresponds to the question representation:
+        **pooler_output**: ``torch.FloatTensor`` of shape ``(batch_size, hidden_size)``
+            Last layer hidden-state of the first token of the sequence (classification token)
+            further processed by a Linear layer. This output is to be used to embed questions for
+            nearest neighbors queries with context embeddings.
+
+    Examples::
+
+        tokenizer = DPRTokenizer.from_pretrained('dpr-base-uncased')
```

Comment:
Add `from transformers import DPRQuestionEncoder, DPRTokenizer ` (we test the docstrings)